Recon/src/Recon/SequentialTrigger.cxx

```diff
@@ -1,54 +1,64 @@
+ * */
+
 void SequentialTrigger::configure(framework::config::Parameters &ps) {
   trigger_list_ = ps.getParameter<std::vector<std::string>>("trigger_list");
   trigger_passNames_ = ps.getParameter<std::vector<std::string>>("trigger_passNames");
-  pass_masks_ = ps.getParameter<std::vector<int>>("pass_mask");
   doAND_ = ps.getParameter<bool>("doAND");
   doOR_ = ps.getParameter<bool>("doOR");
+  doVAL_ = ps.getParameter<bool>("doVAL");
   return;
 }
 
+/**
+ *
+ *This producer takes in a list of triggers and runs an OR or AND skimmer depending on the config file. It will change the keep
+ tag and produce nothing. This is unless the validation command is set to true, in which case it produces a boolean collection in an output file
+ with the keeping flag.
+ *
+ * */
+
 void SequentialTrigger::produce(framework::Event& event) {
   bool hasPassed = false;
   bool keepTrack = false;
-  int  maskValue = 0; 
+  //Returns an error if some bad combination of doOR and doAND is enabled.
+  if (doAND_ and doOR_){
+    ldmx_log(fatal) << "Can't do both doAND and doOR. Exiting.";
+    return;
+  }
+  if (not(doAND_) and not(doOR_)){
+    ldmx_log(fatal) << "Must do one of doAND and doOR. Exiting.";
+    return;
+  }
   for (int i = 0; i<trigger_list_.size(); i++){
-    //if (!event.exists(trigger_list_[i],trigger_passNames_[i])) {
-    //  ldmx_log(fatal) << "Attemping to use non-existing trigger collection "
-    //                  << trigger_list_[i] << "_" << trigger_passNames_[i]
-    //                  << " to skim! Exiting.";
-    //  return;
-    //}
-    if (doAND_ and doOR_){
-      ldmx_log(fatal) << "Can't do both doAND and doOR. Exiting.";
+    //Returns an error is a trigger collection DNE
+    if (!event.exists(trigger_list_[i],trigger_passNames_[i])) {
+      ldmx_log(fatal) << "Attemping to use non-existing trigger collection "
+                      << trigger_list_[i] << "_" << trigger_passNames_[i]
+                      << " to skim! Exiting.";
       return;
     }
     auto trigResult{event.getObject<ldmx::TriggerResult>(trigger_list_[i],trigger_passNames_[i])};
-    //Returns true should any trigger pass and doOR_ enabled; otherwise preps to use masks
+    //Returns true should any trigger pass and doOR_ enabled, and sets keepTrack to true and breaks if doAND_ and any fail
     if(trigResult.passed()){
         if(doOR_){hasPassed=true;break;}
-        maskValue+=(int)pow(2,i);
     }else{
-        keepTrack=true;
+        if(doAND_){keepTrack=true;break;}
     }
   }
   //In the event we have doAND_ set, this will run and only return true if all are true
   if(doAND_){hasPassed=not(keepTrack);}
-  //Uses masks should other methods fail
-  if(not(doOR_)and not(doAND_)){
-    for(int m : pass_masks_){
-        if(maskValue==m){hasPassed=true;break;}
-    }
-  }
+  
   //Used to validate if code was working
-
+  if(doVAL_){
   event.add("validation", hasPassed);
```

Comment:
Does this syntax work? I thought you needed to add a proper collection to the event bus.

Reply:
Yes, this works. You can add anything as simple as a single bool up to a map of user classes to the event bus.

Reply:
🤯